### PR TITLE
fix(codeclimate bug):i was fixing my code climate

### PR DESCRIPTION
**Pull request by Mukunzijames**
what was this PR do

fixing a burge in codeclimate  ,swagger

description

fxing the codeclimate adding new exclude pattern 
adding deployed link swagger documentation

id of the project

#187300135

### CODE REVIEW

**Review by manzidavidrw — APPROVED** [review]

**Review by Sophie20202 — APPROVED** [review]

**Review by TuyishimireEric — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-
+[![CircleCI](https://dl.circleci.com/status-badge/img/gh/atlp-rwanda/alpha-ec-be/tree/develop.svg?style=svg)](https://dl.circleci.com/status-badge/redirect/gh/atlp-rwanda/alpha-ec-be/tree/develop)
```
Comment (fab-ryan) [inline README.md:1]: Why did you change this link

File: src/routers/userRoute.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import validatingUser from '../controllers/loginController';
 
 const router = Router();
 
-router.post('/users', validationMiddleware(userValidationSchema), createUser);
```
Comment (fab-ryan) [inline src/routers/userRoute.ts:9]: Don't change this codes

File: src/swagger.json [hunk 3]
```diff
@@ -9,13 +9,28 @@
...
+        "name":"Authorization",
+        "in":"header"
+      }
+    }
+  },
   "paths": {
     "/api/users/login": {
       "post": {
         "tags": [
-          "login"
```
Comment (fab-ryan) [inline src/swagger.json:18]: Why are changing someone else's codes

File: src/swagger.json [hunk 4]
```diff
@@ -67,7 +82,7 @@
         }
       }
     },
-    "/api/users": {
+    "/api/users/register": {
       "post": {
```
Comment (fab-ryan) [inline src/swagger.json:75]: And again here

File: src/swagger.json [hunk 5]
```diff
@@ -132,41 +147,6 @@
...
-          "name",
-          "email",
-          "phone",
-          "address",
-          "password"
-        ]
-      }
-    }
   }
 }
```
Comment (fab-ryan) [inline src/swagger.json]: You have deleted alot

File: test/user.test.ts [hunk 6]
```diff
@@ -6,7 +6,6 @@ import chaiHttp from 'chai-http';
 import { describe, it } from 'mocha';
 import app from '../src/app';
 import { userValidationSchema } from '../src/validations';
-import { validationMiddleware } from '../src/middleware';
```
Comment (fab-ryan) [inline test/user.test.ts:9]: You have removed this line

File: test/user.test.ts [hunk 7]
```diff
@@ -23,7 +22,7 @@ describe('USER API TEST', () => {
 
     chai
       .request(app)
-      .post('/api/users')
+      .post('/api/users/register')
```
Comment (fab-ryan) [inline test/user.test.ts:25]: Again here why??

**Review by fab-ryan — CHANGES_REQUESTED** [review]
Why did you change much changes

File: src/swagger.json (hunk 3 above)
Comment (Mukunzijames) [reply src/swagger.json:18]: i changed this in order to give login and register same tag and we decided to do it as team

File: src/swagger.json (hunk 5 above)
Comment (Mukunzijames) [reply src/swagger.json:151]: he was trying to show sample and we decided as team to remove it everything is fine

File: README.md (hunk 1 above)
Comment (Mukunzijames) [reply README.md:1]: i added this link it is new that's why it is here

File: .codeclimate.yml [hunk 8]
```diff
@@ -10,3 +10,8 @@ checks:
 exclude_patterns:
   - "**/package-lock.json/"
   - "**/.env/*"
+  - "**/test/*"
+  - "**/tests/*"
+  - "**/vendor/"
+  - "**/*_test.go"
```
Comment (fab-ryan) [inline .codeclimate.yml]: we don't have go languuage

File: src/swagger.json
Comment (fab-ryan) [inline src/swagger.json:1]: you have removed login implementation please don't change somonelse codes

**Review by fab-ryan — CHANGES_REQUESTED** [review]
Change that Please

File: .codeclimate.yml (hunk 8 above)
Comment (fab-ryan) [reply .codeclimate.yml]: you didn't work on my comments

File: src/swagger.json [hunk 9]
```diff
@@ -132,41 +147,6 @@
         }
       }
     }
```
Comment (fab-ryan) [inline src/swagger.json]: You deleted the codes that are mostly needed

**Review by fab-ryan — CHANGES_REQUESTED** [review]
on comments

File: src/swagger.json
Comment (Mukunzijames) [reply src/swagger.json:1]: i have resolved that

File: .codeclimate.yml (hunk 8 above)
Comment (Mukunzijames) [reply .codeclimate.yml]: i removed this

File: src/database/models/role.ts
Comment (fab-ryan) [inline src/database/models/role.ts:1]: why I am seeing this file for roles

Comment (fab-ryan) [reply src/database/models/role.ts:1]: please don't mixtupe
 the tasks

File: tsconfig.json [hunk 10]
```diff
@@ -107,6 +107,6 @@
     "skipLibCheck": true /* Skip type checking all .d.ts files. */
   },
   // "include": ["src/**/*.ts", "src/config/config.js"],
-  "include": ["src/**/*.ts", "src/config/config.js", "test/*/.ts", "swagger.json"],
+  "include": ["src/**/*.ts", "src/config/config.js", "test/*/.ts", "swagger.json", "src/database/models/role.ts"],
```
Comment (fab-ryan) [inline tsconfig.json]: Please remove those `src/database/models/role.ts`

**Review by fab-ryan — CHANGES_REQUESTED** [review]

File: src/database/models/role.ts
Comment (Mukunzijames) [reply src/database/models/role.ts:1]: yes i removed and i really apologize that cause by starting my new task because denis give me approve

File: tsconfig.json (hunk 10 above)
Comment (Mukunzijames) [reply tsconfig.json]: yes i removed that

**Review by fab-ryan — APPROVED** [review]
